### ISSUE TITLE
`dub describe` takes 10-45 seconds every time

### ISSUE THREAD

**Issue by CyberShadow** [dlang/dub#1486]
Editors need to run `dub describe` to enable proper building and code navigation/completion support for D source files. For this to work well, the command needs to execute quickly, otherwise the editor apparently hangs because dub does.

I'm seeing `dub describe` having some pathological execution times (and additionally producing bogus results) on a flaky connection:

```
~/work/DFeed » time dub describe
Package ae not found in registry at https://code.dlang.org/ (fallback ["registry at http://code.dlang.org/", "registry at https://code-mirror.dlang.io/", "registry at https://code-mirror2.dlang.io/", "registry at https://dub-registry.herokuapp.com/"]): Timeout was reached on handle 55FD19CC3B20
Root package dfeed:lib reference ae 0.0.2190 cannot be satisfied.
Packages causing the conflict:
	dcaptcha depends on >=0.0.1969
	dfeed:lib depends on 0.0.2190
	dfeed:lib depends on 0.0.2139
	dfeed:lib depends on 0.0.2139
dub describe  0.04s user 0.04s system 0% cpu 44.244 total
─────────────────────────────────────[ master* │ 2:usage │ 2018-06-02 19:48:16.664 ]──
~/work/DFeed » time dub describe
Root package dfeed:lib reference ae 0.0.2190 cannot be satisfied.
Packages causing the conflict:
	dcaptcha depends on >=0.0.1969
	dfeed:lib depends on 0.0.2190
	dfeed:lib depends on 0.0.2139
	dfeed:lib depends on 0.0.2139
dub describe  0.06s user 0.05s system 1% cpu 10.286 total
─────────────────────────────────────[ master* │ 2:usage │ 2018-06-02 19:48:43.736 ]──
~/work/DFeed » time dub describe
Root package dfeed:lib reference ae 0.0.2190 cannot be satisfied.
Packages causing the conflict:
	dcaptcha depends on >=0.0.1969
	dfeed:lib depends on 0.0.2190
	dfeed:lib depends on 0.0.2139
	dfeed:lib depends on 0.0.2139
dub describe  0.07s user 0.04s system 0% cpu 17.706 total
─────────────────────────────────────[ master* │ 2:usage │ 2018-06-02 19:49:08.415 ]──
~/work/DFeed » time dub describe
Root package dfeed:lib reference ae 0.0.2190 cannot be satisfied.
Packages causing the conflict:
	dcaptcha depends on >=0.0.1969
	dfeed:lib depends on 0.0.2190
	dfeed:lib depends on 0.0.2139
	dfeed:lib depends on 0.0.2139
dub describe  0.08s user 0.02s system 0% cpu 12.798 total
```

There are a few problems here:

- Ideally, `dub describe` should not involve "phoning home" implicitly. This ought to be explicit.
- In situations where the invocation is not done directly by a human, network failures should probably be cached (for a short duration) in a similar way as successful requests, to prevent successive invocations from resulting in more pointless hangs.
- The timeouts are too generous. 45 seconds is much too long.
- The "reference ae 0.0.2190 cannot be satisfied" part is bogus - the specified version clearly is registered with code.dlang.org: https://code.dlang.org/packages/ae/0.0.2190
  It's possible that the message is masking a network error.

The most practical fix for this situation would probably be to fix or configure the editor integration to use `--skip-registry=all`, but I'm still going to file this as it looks there are plenty of actionables on Dub's side.

**Comment by WebFreak001** [comment]
I agree it's very slow for packages with lots of dependencies and something definitely needs to be done about this. Also you might want to consider using dub as a library.

**Comment by CyberShadow** [comment]
> The "reference ae 0.0.2190 cannot be satisfied" part is bogus - the specified version clearly is registered with code.dlang.org: https://code.dlang.org/packages/ae/0.0.2190

Oops, on closer look dub is right here, there is an unsatisfiable version conflict:
https://github.com/CyberShadow/DFeed/commit/27c68007e616f4738f46c9b08f2a5d0055a8fdf5

**Comment by SingingBush** [comment]
I just timed dub describe on a project of mine and found that the first attempt was just short of 60 seconds whereas subsequent calls to dub describe were all less than a second. Adding `--skip-registry=all` didn't seem to make an impact.

When [Intellij-DLanguage](https://github.com/intellij-dlanguage/intellij-dlanguage/) runs dub describe it's basically after dependency info so we can attach the dependency sources to the project. It would potentially be helpful to have a command that just gets the info that an IDE would need as fast as possible.
